developer/src/kmcmplib/src/Compiler.cpp

```diff
@@ -284,14 +284,17 @@ KMX_BOOL AddCompileError(KMX_DWORD msg)
     strcpy(szText, szTextp);
   }
   else {
-    sprintf(szText, "Unknown error %x", msg);
+    snprintf(szText, 793, "Unknown error %x", msg);
```

Comment:
Let's fix it with proper constants:

```c++
#define COMPILE_ERROR_MAX_LEN (SZMAX_ERRORTEXT + 1 + 280)
```

Then, fix the def of `szText` at the top of the function, and:

```suggestion
    snprintf(szText, COMPILE_ERROR_MAX_LEN, "Unknown error %x", msg);
```

Reply:
done